### PR TITLE
SONAR-9346 delete links only for root

### PR DESCRIPTION
**Pull request by sns-seb**
base on #2127

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/component/ComponentFinder.java [hunk 1]
```diff
@@ -41,53 +41,84 @@
...
+    checkByUuidOrKey(componentUuid, componentKey, parameterNames);
+
+    if (componentUuid != null) {
+      return checkIsProject(getByUuid(dbSession, checkParamNotEmpty(componentUuid, parameterNames.getUuidParam()), LABEL_PROJECT));
+    }
+
+    return checkIsProject(getByKey(dbSession, checkParamNotEmpty(componentKey, parameterNames.getKeyParam()), LABEL_PROJECT));
+  }
+
+  private String checkParamNotEmpty(String value, String param) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/component/ComponentFinder.java]: MAJOR Make "checkParamNotEmpty" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInProjectLinks.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  }
+
+  @Override
+  protected void execute(Context context) throws SQLException {
+    MassUpdate massUpdate = context.prepareMassUpdate();
+    massUpdate.select("select" +
+        " pl.id" +
+        " from project_links pl" +
+        " where" +
+        " not exists (select 1 from projects p where p.uuid = pl.component_uuid and p.scope = ? and p.qualifier in (?,?))")
```
Comment (simonbrandhof) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInProjectLinks.java]: you could add `and p.enabled = true`

Comment (sns-seb) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInProjectLinks.java]: done

File: server/sonar-server/src/main/java/org/sonar/server/component/ComponentFinder.java (hunk 1 above)
Comment (sns-seb) [inline server/sonar-server/src/main/java/org/sonar/server/component/ComponentFinder.java]: done

**Pull request merged**